aloha-io-avro/src/main/scala/com/eharmony/aloha/audit/impl/avro/AvroGenericRecordAuditor.scala

```diff
@@ -0,0 +1,215 @@
+package com.eharmony.aloha.audit.impl.avro
+
+import com.eharmony.aloha.audit.MorphableAuditor
+import com.eharmony.aloha.id.ModelIdentity
+import com.eharmony.aloha.reflect.{RefInfo, RefInfoOps}
+import org.apache.avro.Schema
+import org.apache.avro.Schema.Type.RECORD
+import org.apache.avro.generic.{GenericData, GenericRecord}
+import org.apache.commons.io.IOUtils
+import java.{util => ju}
+import scala.collection.JavaConversions.{seqAsJavaList, collectionAsScalaIterable}
+
+/**
+  * An auditor for encoding information in a tree structure as an Avro `org.apache.avro.generic.GenericRecord`.
+  *
+  * See the resources in [[com.eharmony.aloha.audit.impl.avro]] for the Avro schema.
+  *
+  * Created by ryan on 2/26/17.
+  *
+  * @tparam N the natural output type of a model whose data is to be audited.
+  */
+sealed abstract class AvroGenericRecordAuditor[N]
+  extends MorphableAuditor[GenericRecord, N, GenericRecord]
+     with Serializable
+
+object AvroGenericRecordAuditor extends Serializable {
+  /**
+    * '''NOTE''': The class loader is used to locate the Avro schema specification rather than
+    * Apache VFS 1 or 2.  This is done because issues can arise when users initialize both
+    * VFS 1 and 2.  Since the choice of auditor shouldn't affect the user's choice or VFS
+    * version elsewhere in Aloha, we don't use VFS.
+    */
+  private[this] val SchemaResourceLocation = "com/eharmony/aloha/audit/impl/avro/generic_record_auditor.avsc"
+
+  // TODO: Should the schemas be inside AvroGenericRecordAuditor instead of in the companion object?
```

Comment:
Not right now.  Let's do this [https://github.com/julianpeeters/avrohugger](avro hugger) stuff later and create a new score that provides a compatible on-the-wire data.  Are you OK with that?